tests/unit_tests/test_message_queue_service.py

```diff
@@ -0,0 +1,86 @@
+import asyncio
+
+import aio_pika
+import pytest
+
+import mock
+from server.config import config
+from server.decorators import with_logger
+from server.message_queue_service import MessageQueueService
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture
+async def mq_service():
+    service = MessageQueueService()
+    await service.initialize()
+
+    await service.declare_exchange("test_exchange")
+
+    yield service
+
+    await service.shutdown()
+
+
+async def test_initialize():
+    service = MessageQueueService()
+    await service.initialize()
+    await service.shutdown()
+
+
+async def test_publish(mq_service):
+    payload = {"msg": "test message"}
+    exchange_name = "test_exchange"
+    routing_key = "test.routing.key"
+    delivery_mode = aio_pika.DeliveryMode.NOT_PERSISTENT
+
+    await mq_service.publish(exchange_name, routing_key, payload, delivery_mode)
+
+
+async def test_incorrect_credentials(mocker):
```

Comment:
I think you can use the `caplog` fixture to assert that certain warning messages appear in the log

Reply:
Then you don't need to mock the logger

Reply:
I don't think that exists in the current stable pytest version yet (5.4.3)?

Reply:
Well the `caplog` fixture does exist, I'm using it in some of the tests to change the log level. I haven't tried to assert contents of log messages though

Reply:
Yes, I noticed, I just followed the link and then clicked on the "latest stable version" banner, which tries to get you to the logging page of 5.4.3 which doesn't exist, so I wrongly assumed that's because they added that feature.
Just changed it, works like a charm.